packages/common/lib/common/password.js

```diff
@@ -81,19 +81,27 @@ const password = pw => {
                 return callback('Unknown');
             }
 
-            this.hash(key[3], parseInt(key[1], 10), function (error, newHash) {
+            this.hash(key[3], parseInt(key[1], 10), (error, newHash) => {
                 if (error) {
-                    return callback(error);
+                    callback(error);
+                } else {
+                    callback(null, newHash === hashedPassword);
                 }
-                callback(null, newHash === hashedPassword);
             });
-
         },
-        complexity: _callback => {
-            // Todo: Check for password complexity
-            return true;
+        complexity: (password, callback) => {
+            let result = false;
+            if (typeof password === 'string') {
+                result =
+                    password.length >= 8 &&           // minimum length is 8
+                    password.match(/\d/) &&    // contains at least one digit
+                    password.match(/[a-z]/) && // contains at least one lower case letter
+                    password.match(/[A-Z]/);   // contains at least one upper case letter
+            }
+            typeof callback === 'function' && callback(result);
+            return result; // true if the complexity OK
```

Comment:
Cb and return really needed?

Reply:
I assume because both ways used somehwere? @GermanBluefox